src/app/modules/common/dual-list/dual-list.component.ts

```diff
@@ -45,6 +47,12 @@ export class DualListboxComponent<T extends { id: string | number; name?: string
     this.selectedItems = new ListSelectionImpl(this._selectedItems);
   }
 
+  ngOnChanges(changes: IxSimpleChanges<this>): void {
+    if ('items' in changes && !_.isEqual(changes.items.currentValue, changes.items.previousValue)) {
+      this.ngOnInit();
```

Comment:
Calling one lifecycle hook in another doesn't look right. It's better to extract code in a separate method and call it.